components/list/list-item.js

```diff
@@ -0,0 +1,183 @@
+			::slotted(.d2l-list-item-text-secondary) {
+				margin: 0;
+				margin-top: 0.3rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: none;
+			}
+		`;
+		const divider = css`
+			.d2l-list-item-divider-top,
+			.d2l-list-item-divider-bottom {
+				border: 0;
+				border-top: 1px solid var(--d2l-color-mica);
+				margin: 0;
+				position: absolute;
+				width: 100%;
+			}
+			.d2l-list-item-divider-top {
+				display: var(--d2l-list-item-divider-top, none);
+				top: -1px;
+			}
+			.d2l-list-item-divider-bottom {
+				bottom: -1px;
+				display: var(--d2l-list-item-divider-bottom, none);
+			}
+		`;
+
+		const breakPoint580 = css`
+			.d2l-list-item-flex[breakpoint="580"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 71px;
+				max-width: 120px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="580"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="580"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+
+		const breakPoint636 = css`
+			.d2l-list-item-flex[breakpoint="636"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 102px;
+				max-width: 180px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="636"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="636"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+
+		const breakPoint842 = css`
+			.d2l-list-item-flex[breakpoint="842"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 120px;
+				max-width: 216px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="842"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="842"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+		return [ layout, mainContent, divider, breakPoint580, breakPoint636, breakPoint842];
+	}
+
+	constructor() {
+		super();
+		this._breakpoint = 0;
+		this._breakpointList = [842, 636, 580, 0];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-list-item-flex" breakpoint="${this._breakpoint}">
+				<slot name="illustration-outer"></slot>
+				<div class="d2l-list-item-content">
+					<hr class="d2l-list-item-divider-top"></hr>
+					<div class="d2l-list-item-content-flex">
+						<slot name="illustration"></slot>
+						<div class="d2l-list-item-main"><slot></slot></div>
+						<slot name="actions"></slot>
+					</div>
+					<hr class="d2l-list-item-divider-bottom"></hr>
+				</div>
+			</div>
+		`;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		ro.observe(this);
+	}
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		ro.unobserve(this);
+	}
+	resizedCallback(rect) {
```

Comment:
Using this resize callback to activate the breakpoints. This is important because the responsive behaviour is dependant on the width of the component and not the width of the screen. So `@media` won't work at all.

Reply:
It doesn't but I just realized that this would be so much better if I can put this in `d2l-list`. Not sure how but it would be nice.

Reply:
This is used to change the style that corresponds to the different breakpoints.